### PR TITLE
cbqn: 2023-02-01 -> 0.2.0

### PR DESCRIPTION
**Pull request by Detegr**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/bqn/cbqn/default.nix [hunk 1]
```diff
@@ -95,6 +95,9 @@ stdenv.mkDerivation rec {
     install -Dm644 include/bqnffi.h -t "$dev/include"
     install -Dm755 libcbqn${stdenv.hostPlatform.extensions.sharedLibrary} -t "$lib/lib"
   ''
+  + lib.optionalString stdenv.hostPlatform.isDarwin ''
+    install_name_tool -id $lib/lib/libcbqn.dylib $lib/lib/libcbqn.dylib
```
Comment (sternenseemann) [inline pkgs/development/interpreters/bqn/cbqn/default.nix:99]: Should be possible to use fixDarwinDylibNames for this instead.

**Review by sternenseemann — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/bqn/cbqn/default.nix (hunk 1 above)
Comment (Detegr) [reply pkgs/development/interpreters/bqn/cbqn/default.nix]: I wasn't aware of `fixDarwinDylibNames`. Fixed and seems to work.

File: pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix [hunk 2]
```diff
@@ -5,13 +5,13 @@
 
 stdenvNoCC.mkDerivation {
   pname = "cbqn-bytecode";
-  version = "unstable-2023-01-27";
+  version = "unstable-2023-04-19";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix:9]: Split to another commit

File: pkgs/development/interpreters/bqn/cbqn/default.nix [hunk 3]
```diff
@@ -103,7 +104,7 @@ stdenv.mkDerivation rec {
     homepage = "https://github.com/dzaima/CBQN/";
     description = "BQN implementation in C";
     license = licenses.gpl3Plus;
-    maintainers = with maintainers; [ AndersonTorres sternenseemann synthetica shnarazk ];
+    maintainers = with maintainers; [ AndersonTorres sternenseemann synthetica shnarazk detegr ];
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/bqn/cbqn/default.nix:107]:
```suggestion
    maintainers = with maintainers; [ AndersonTorres sternenseemann synthetica shnarazk ];
```

Makes no sense to maintain only the interpreter but not the bytecode.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix (hunk 2 above)
Comment (sternenseemann) [reply pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix:9]: Why? cbqn and bytecode should be updated together, atomically.

File: pkgs/development/interpreters/bqn/cbqn/default.nix (hunk 3 above)
Comment (Detegr) [reply pkgs/development/interpreters/bqn/cbqn/default.nix:107]: Added myself as maintainer for bytecode package as well.

File: pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix (hunk 2 above)
Comment (Detegr) [reply pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix:9]: As @sternenseemann said, `cbqn-bytecode` should match the submodule used by `CBQN`. If done in two commits, the first commit will result in a version that either doesn't build correctly or doesn't work correctly.

Comment (AndersonTorres) [reply pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix:9]: Also, how exactly this will mismatch? Is this a serious argument?
 
- Ofborg will build both "atomically"; there is no risk of mismatch

- When merged, the end users will see the ending result "atomically", not the intermediary steps; there is no risk of mismatch

- It is not uncommon to include multiple commits in a same PR. Indeed this is the typical practice when a program requires a NixOS module and a test suite. Example: kavita

- If the "atomic jumbo commit" made sense, then cimg, gmic and gmic-qt should be a jumbo commit, but they obviously aren't:

**Review by shnarazk — APPROVED** [review]

File: pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/development/interpreters/bqn/cbqn/cbqn-bytecode.nix:9]: The magic words are `git bisect` and `git cherry-pick`. 
A more granular git history is more helpful to track changes.

**Review by AndersonTorres — APPROVED** [review]

**Review by shnarazk — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 229433` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>cbqn-replxx</li>
    <li>cbqn-replxx.dev</li>
    <li>cbqn-replxx.lib</li>
    <li>cbqn-standalone-replxx</li>
    <li>cbqn-standalone-replxx.dev</li>
    <li>cbqn-standalone-replxx.lib</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>cbqn</li>
    <li>cbqn-standalone</li>
    <li>cbqn-standalone.dev</li>
    <li>cbqn-standalone.lib</li>
    <li>cbqn.dev</li>
    <li>cbqn.lib</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build cbqn-replxx

**Comment by AndersonTorres** [comment]
Relevant (?) line:

> Error: 'shared' and 'REPLXX' cannot both be enabled

**Comment by Detegr** [comment]
Seems enabling both `shared` and `REPLXX` used to work, but has become an error in the newer versions.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 229433` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>cbqn</li>
    <li>cbqn-replxx</li>
    <li>cbqn-standalone</li>
    <li>cbqn-standalone-replxx</li>
    <li>cbqn-standalone.dev</li>
    <li>cbqn-standalone.lib</li>
    <li>cbqn.dev</li>
    <li>cbqn.lib</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Review by shnarazk — APPROVED** [review]
